src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -48,8 +50,17 @@ protected override void Load(ContainerBuilder builder)
 
                 return logger;
             })
-            .As<ILogger>()
-            .ExternallyOwned();
+            .As<ILogger>();
+
+            if (_dispose)
+            {
+                registration.SingleInstance()
+                    .OnRelease(logger => Log.CloseAndFlush());
```

Comment:
This shouldn't be necessary, since Autofac will dispose `logger` whether it is the static/global logger, or a specific one passed as a parameter. (`Log.CloseAndFlush()` is just a shortcut for `Log.Logger.Dispose()`.)